File: dbflow/src/main/java/com/raizlabs/android/dbflow/runtime/TableNotifierRegister.java
package com.raizlabs.android.dbflow.runtime;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Description: Defines how {@link ModelNotifier} registers listeners. Abstracts that away.
 */
public interface TableNotifierRegister {

    <T> void register(@NonNull Class<T> tClass);

    <T> void unregister(@NonNull Class<T> tClass);

    void unregisterAll();

    void setListener(@Nullable OnTableChangedListener listener);

    boolean isSubscribed();
}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/runtime/TableNotifierRegister.kt
package com.raizlabs.android.dbflow.runtime

/**
 * Description: Defines how [ModelNotifier] registers listeners. Abstracts that away.
 */
interface TableNotifierRegister {

    val isSubscribed: Boolean

    fun <T> register(tClass: Class<T>)

    fun <T> unregister(tClass: Class<T>)

    fun unregisterAll()

    fun setListener(listener: OnTableChangedListener?)
}
